public/app/features/plugins/extensions/registryFactory.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  AppConfigureExtension,
+  AppPluginExtensionLink,
+  AppPluginExtensionLinkConfig,
+  PluginExtensionLink,
+  PluginExtensionTypes,
+} from '@grafana/data';
+import type {
+  PluginExtensionRegistry,
+  PluginExtensionRegistryItem,
+  RegistryConfigureExtension,
+} from '@grafana/runtime';
+
+import { PluginPreloadResult } from '../pluginPreloader';
+
+import { createErrorHandling } from './errorHandling';
+import { createLinkValidator, isValidLinkPath } from './validateLink';
+
+export function createPluginExtensionRegistry(preloadResults: PluginPreloadResult[]): PluginExtensionRegistry {
+  const registry: PluginExtensionRegistry = {};
+
+  for (const result of preloadResults) {
+    const limiter: Record<string, number> = {};
```

Comment:
**Nit:** maybe something like this would communicate the intent better?
```suggestion
    const pluginPlacementCount: Record<string, number> = {};
```

Reply:
Agreed. I've pushed a commit to rename this. 👍